runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -519,24 +518,27 @@ private void cleanUpForRetry() {
 
         // Re ask for the Head, if it changes while we were trying.
         findAndSetLogHead();
-
+        findAndSetLogTail();
```

Comment:
Should we add a validation that the head and tail chosen are chosen on the same epoch?
Probably this is already taken care of by the inherent order of setting the head before the tail, but I think it would be safer to put this validation here.